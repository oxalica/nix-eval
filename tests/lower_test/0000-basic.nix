{
  hello = "world";
  paths = [ <nix> ./nix /nix http://nix ];
  cond = if 0 == 0 then 42 else 4.2;
  foo = {}.a or {}.b;
  buildPhase = ''
    echo "${"hello" + "world"}"
  '';
}
