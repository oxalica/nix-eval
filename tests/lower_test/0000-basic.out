Ok(
    Expr(
        AttrSet {
            entries: {
                "buildPhase": Expr(
                    Str {
                        parts: [
                            Literal(
                                "echo ",
                            ),
                            Expr(
                                Expr(
                                    BinOp {
                                        operator: Add,
                                        lhs: Expr(
                                            Literal(
                                                String(
                                                    "hello",
                                                ),
                                            ),
                                        ),
                                        rhs: Expr(
                                            Literal(
                                                String(
                                                    "world",
                                                ),
                                            ),
                                        ),
                                    },
                                ),
                            ),
                            Literal(
                                "",
                            ),
                        ],
                    },
                ),
                "cond": Expr(
                    IfElse {
                        condition: Expr(
                            BinOp {
                                operator: Equal,
                                lhs: Expr(
                                    Literal(
                                        Int(
                                            0,
                                        ),
                                    ),
                                ),
                                rhs: Expr(
                                    Literal(
                                        Int(
                                            0,
                                        ),
                                    ),
                                ),
                            },
                        ),
                        then_body: Expr(
                            Literal(
                                Int(
                                    42,
                                ),
                            ),
                        ),
                        else_body: Expr(
                            Literal(
                                Float(
                                    4.2,
                                ),
                            ),
                        ),
                    },
                ),
                "foo": Expr(
                    Select {
                        set: Expr(
                            AttrSet {
                                entries: {},
                                dynamics: [],
                            },
                        ),
                        index: Expr(
                            Literal(
                                String(
                                    "a",
                                ),
                            ),
                        ),
                        or_default: Some(
                            Expr(
                                Select {
                                    set: Expr(
                                        AttrSet {
                                            entries: {},
                                            dynamics: [],
                                        },
                                    ),
                                    index: Expr(
                                        Literal(
                                            String(
                                                "b",
                                            ),
                                        ),
                                    ),
                                    or_default: Some(
                                        Expr(
                                            Select {
                                                set: Expr(
                                                    AttrSet {
                                                        entries: {},
                                                        dynamics: [],
                                                    },
                                                ),
                                                index: Expr(
                                                    Literal(
                                                        String(
                                                            "c",
                                                        ),
                                                    ),
                                                ),
                                                or_default: None,
                                            },
                                        ),
                                    ),
                                },
                            ),
                        ),
                    },
                ),
                "hello": Expr(
                    Literal(
                        String(
                            "world",
                        ),
                    ),
                ),
                "paths": Expr(
                    List {
                        items: [
                            Expr(
                                Literal(
                                    Path(
                                        Store,
                                        "nix",
                                    ),
                                ),
                            ),
                            Expr(
                                Literal(
                                    Path(
                                        Relative,
                                        "./nix",
                                    ),
                                ),
                            ),
                            Expr(
                                Literal(
                                    Path(
                                        Absolute,
                                        "/nix",
                                    ),
                                ),
                            ),
                            Expr(
                                Literal(
                                    String(
                                        "http://nix",
                                    ),
                                ),
                            ),
                        ],
                    },
                ),
            },
            dynamics: [],
        },
    ),
)